testsuite/integration/manualmode/src/test/java/org/wildfly/test/manual/microprofile/health/MicroProfileHealthDefaultEmptyStartupTestBase.java

```diff
@@ -0,0 +1,295 @@
+    private ContainerController containerController;
+
+    @Test
+    @InSequence(1)
+    public void testApplicationStartupWithoutDeploymentWithDefaultEmptyStartup() throws Exception {
+
+        final CompletableFuture<Void> stringCompletableFuture = CompletableFuture.runAsync(() -> {
+            boolean connectionEstablished = false;
+
+            do {
+                try {
+                    checkGlobalOutcome(managementClient, "check-started", false, null);
+                    connectionEstablished = true;
+                } catch (ConnectException e) {
+                    // OK, server is not started yet
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+
+            } while (!connectionEstablished);
+
+            try {
+                checkGlobalOutcome(managementClient, "check-started", false, null);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
+
+        Map<String, String> map = getVMArgMap(DISABLE_DEFAULT_PROCEDURES_PROPERTY);
+        containerController.start(CONTAINER_NAME, map);
+
+        try {
+            stringCompletableFuture.get(1, TimeUnit.MINUTES);
+        } finally {
+            containerController.stop(CONTAINER_NAME);
+        }
+    }
+
+    @Test
+    @InSequence(2)
+    public void testApplicationStartupWithEmptyDeploymentWithDefaultEmptyStartup() throws Exception {
+
+        // deploy the application and stop the container
+        containerController.start(CONTAINER_NAME);
+        deployer.deploy(MICRO_PROFILE_HEALTH_APPLICATION_WITHOUT_STARTUP_TEST_BASE);
+        containerController.stop(CONTAINER_NAME);
+
+        // check that the startup status is changed to UP once the user deployment checks (which there are none
+        // in this case) are processed
+        final CompletableFuture<Void> stringCompletableFuture = CompletableFuture.runAsync(() -> {
+            boolean connectionEstablished = false;
+
+            do {
+                try {
+                    checkGlobalOutcome(managementClient, "check-started", false, null);
+                    connectionEstablished = true;
+                } catch (ConnectException e) {
+                    // OK, server is not started yet
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                } catch (ComparisonFailure cf) {
+                    // this might happen if the first call hits already processed user checks (responds UP)
+                    // so continue to check if the rest of the responses is correct
+                    connectionEstablished = true;
+                }
+
+            } while (!connectionEstablished);
+
+            boolean userChecksProcessed = false;
+
+            do {
+                try {
+                    checkGlobalOutcome(managementClient, "check-started", false, null);
+                } catch (AssertionError ae) {
```

Comment:
the problem with this that we would need to rely on the message relayed from JUnit which I didn't want to depend on. The next check should fail if something would be met.

Reply:
Hi @xstefank - I've been thinking about this and I see several assertions could fail in the overridden `checkGlobalOutcome` methods.
Each time this would happen, here, this test method logic would interpret that as "ok, user checks have been processed, let's do the real request now".

I see a risk in this when thinking - for instance - about a HTTP 404 which could happen for some reason, let's imagine - just for example - a of change in the spec which affects the endpoint URL, which has not yet been reflected here in the test.

What would happen then? This is the main question IMHO. 
If the answer is "nothing, because the following assertions or calls would fail and we would get to the actual root cause quickly", well that's ok and we can consider that logic as safe.
Instead, if the following assertions should work, then the test outcome could be affected by this wrong evaluation. And the logic wouldn't be so "safe".

I might be overthinking it, just let me know what your thoughts are, since you know the server code better than me, and I'll be glad to take your word for the final decision.

On a side note, maybe a solution could be to guard the assertions from causing misinterpretation, for instance the HTTP response status code could be checked in order to be sure it is [200|503] (as per the specs) and throw a specific exception which could be recognized in the outer scope, as it happens now for `IOException`.

But, as said, I don't want to insist on this, _provided_ that we're safe with the test method business logic and don't get any issue because of that.

Reply:
it's not totally clear I understand, but I didn't want to plug that much into possible refactoring of duplicated code later. But I agree this can be changed and I will throw a custom exception if there is an invalid response that can be caught.

Reply:
Thanks for addressing my last comment about using the exception, which IMHO offers less opportunity for misinterpretation by the test business logic.

As per https://issues.redhat.com/browse/WFLY-15114, these classes logic will be refactored in the future and while doing that there could be the chance to improve tit further for yet more granularity.
I think this is fine now, and the thread can be resolved.